### DISCUSSION TITLE
Heartbeat Callback

### DISCUSSION THREAD

**Discussion by ravi-kotha** [Farfetch/kafkaflow#163] [General]
Hi,

Is there a way to listen into a callback that relays the kafka consumer heartbeat to the application that uses the kafka flow?
Wanted to use this heartbeat to decide the health of the application so we can scale accordingly.

Thanks,
Ravi

**Comment by filipeesch** [comment]
Hi @ravi-kotha,

You can register a custom consumer factory in the setup process and there you can do the health check using a decorator design pattern. Another way is to create a middleware that catches any exception.

**Reply by ravi-kotha** [reply]
Hey @filipeesch, Do you have any examples for these?

**Reply by filipeesch** [reply]
Unfortunally, not yet. :(
